src/main/java/br/edu/ufape/agiota/negocio/services/ParcelaService.java

```diff
@@ -24,8 +24,23 @@ public List<Parcela> listarParcelasPorEmprestimo(long emprestimoId) {
 
     @Override
     public Parcela buscarParcela(long id) {
-        Optional<Parcela> parceOpt = parcelaRepository.findById(id);
-        if (parceOpt.isPresent()) return parceOpt.get();
+        Optional<Parcela> parcelaOpt = parcelaRepository.findById(id);
+        if (parcelaOpt.isPresent()) {
+            return parcelaOpt.get();
+        }
         throw new RegistroNaoEncontradoException("Parcela com o identificador " + id + " não foi encontrada!");
     }
+
+    @Override
+    public Parcela buscarParcela(long id, long emprestimoId) {
+        Optional<Parcela> parcelaOpt = parcelaRepository.findById(id)
+            .filter(parcela -> parcela.getEmprestimo().getId() == emprestimoId);
+
+        if (parcelaOpt.isPresent()) {
+            return parcelaOpt.get();
+        }
+
+        throw new RegistroNaoEncontradoException("Parcela com o identificador " + id + 
+                                                 " não foi encontrada no empréstimo " + emprestimoId + "!");
+    }
```

Comment:
Método não utilizado. Removê-lo caso não haja necessidade.
Se for usar no futuro, então o método do repository deve ser refatorado para receber o id do empréstimo também, fazendo a consulta diretamente no banco em vez de filtrar no código.